### PR TITLE
Stream batch produce consume

### PR DESCRIPTION
**Pull request by StevenPG**
Adding a clearer example as suggested by @sobychacko in https://github.com/spring-cloud/spring-cloud-stream-samples/pull/240.

This addition provides a concrete working example of using Batch Consumers and Batch Producers as documented [here](https://docs.spring.io/spring-cloud-stream/docs/current/reference/html/spring-cloud-stream.html#_batch_consumers)

Let me know if anything isn't up to Spring Cloud standards or any updates need to be made!

### CODE REVIEW

File: samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@SpringBootApplication
+public class KafkaStreamFunctionBatch {
```
Comment (sobychacko) [inline samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java:32]: I would suggest renaming from `KafkaStream...` as that might get confused with Kafka Streams.

File: samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
```
Comment (sobychacko) [inline samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java:30]: Need author tag.

**Review by sobychacko — CHANGES_REQUESTED** [review]

File: samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java (hunk 2 above)
Comment (StevenPG) [reply samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java]: Added in fb877f75eb919c491422e6cdc4f3b456a0b28f4c

File: samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java (hunk 1 above)
Comment (StevenPG) [reply samples/stream-function-batch-producer-consumer/src/main/java/sample/stream/batch/KafkaStreamFunctionBatch.java]: Updated in fb877f75eb919c491422e6cdc4f3b456a0b28f4c

**Review by sobychacko — APPROVED** [review]

**Comment by sobychacko** [comment]
@StevenPG Apologies for the delay in merging the PR. It looks good. Merged upstream after squashing the commits. I shortened the commit message a bit. Thanks for the PR! More contributions are welcomed.

**Pull request merged**